includes/class-main.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * Plugin Name:       Coding Challenge
+ * 
+ * @package Coding_Challenge
+ * @subpackage Coding_Challenge_main_class
+ * @since 0.1.0 ( plugin creation)
+ * @author mikesm1118
+ * 
+ * @wordpress-plugin 
+ * Plugin Name:       Coding Challenge
+ * Plugin URI:        https://wordpress.org/plugins/coding-challenge
+ * Description:       This plugin creates an admin menu for changing the website's background color.
+ * Version:           0.1.0
+ * Requires at least: 5.1
+ * Requires PHP:      5.6
+ * Author:            mikesm1118
+ * Author URI:        https://profiles.wordpress.org/mikesm1118/
+ * License:           GPL v2 or later
+ * License URI:       https://www.gnu.org/licenses/gpl-2.0.html
+ * Text Domain:       coding-challenge
+ * Domain Path:       /languages
+ */
+
+/**
+ * Class:  Coding_Challenge
+ * 
+ * Main class and activation hooks. 
+ *
+ * @link https://wordpress.org/plugins/coding-challenge
+ *
+ * @package Coding_Challenge
+ * @subpackage Coding_Challenge_main_class
+ * @since 0.1.0 ( plugin creation)
+ */
+class Coding_Challenge {
+
+	/** 
+	 * Constructor.
+	 * 
+	 * @since 0.1.0
+	 * 
+	 */
+	public function __construct() {
+		$this->load_plugin_files();
+		$this->admin = new Coding_Challenge_Admin( $this );
```

Comment:
$this->admin property needs to be declared as a class property before assigning it a value.

Reply:
class property declared